docs/developer-guide/api-overview.md

```diff
@@ -20,153 +20,154 @@ The relevant modules are available in Maven Central:
+  available.
 
-* The `maxContentBytes` sets an upper limit for the message 
-  body, protecting Styx from exhausting memory in face of very long message
-  bodies, or perhaps from denial of service attacks. If the limit is reached,
-  the conversion fails with `ContentOverflowException`. 
+* The `maxContentBytes` sets an upper limit for the aggregated message 
+  body. This acts as a safety valve protecting Styx from exhausting heap in face 
+  of very long message bodies, or perhaps from denial of service attacks. 
+  If the limit is reached, the conversion fails with `ContentOverflowException`. 
 
-Converting a full message to a stream is much easier. The `FullHttpRequest` 
-(`FullHttpResponse`) has a method called `toStreamingRequest` (or `toStreamingResponse`): 
+Call the `stream` method to convert an immutable `HttpRequest` 
+(`HttpResponse`) to a `LiveHttpRequest` (`LiveHttpResponse`):
+ 
 ```java
-    public HttpRequest toStreamingRequest()
+    public LiveHttpRequest stream()
 ``` 
 
-The method signature reveals that `HttpResponse` is available immediately.
-Because the content is permanently stored in full, you can clone a
-full message object into as many streaming objects as necessary. For example:
+The method signature reveals that `LiveHttpResponse` is available immediately.
+Because the content is permanently stored in full, you can clone an immutable
+message object into as many streaming objects as necessary. For example:
 
 ```java
     public class PingHandler extends BaseHttpHandler {
-        private static final FullHttpResponse PONG = response(OK)
+        private static final HttpResponse PONG = response(OK)
                 .disableCaching()
                 .contentType(PLAIN_TEXT_UTF_8)
                 .body("pong", UTF_8)
                 .build();
         
         @Override
-        protected HttpResponse doHandle(HttpRequest request) {
-            return PONG.toStreamingResponse();
+        protected LiveHttpResponse doHandle(LiveHttpRequest request) {
+            return PONG.stream();
         }
     }   
 ```
  
 ### Http Interceptor Interface
 
-A HTTP interceptor is an object that transforms, responds, or runs side-effecting actions 
-for HTTP traffic passing through. 
-
-Styx arranges interceptors in a linear chain, forming a core of its proxying pipeline.
-All recieved traffic goes through the interceptor chain which then acts on the traffic
-accordingly.  
+Styx has a HTTP pipeline that processes all received HTTP messages.
+The pipeline is made of a linear chain of HTTP interceptors. An HTTP interceptor 
+transforms, responds, or runs side-effecting actions on a live HTTP traffic.
   
-Styx has some internal Styx interceptors. But normally it is the custom plugins that
-add value for Styx deployments. An extension point for a custom plugin is the 
-`HttpInterceptor` interface. It has only one method:
+Styx provides some internal interceptors. You can also implement own
```

Comment:
Rewrite as:

Styx provides some internal interceptors. You can also implement your own
interceptor plugins. An extension point for a 
custom plugin is the `HttpInterceptor` interface. It has only one method:

(Left out "That's what Styx is for!" - if we want to emphasise Styx's value as a plugin platform we can do so elsewhere).

Reply:
👍